opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1003,13 +983,11 @@ namespace Opm {
                 const auto& phaseName = FluidSystem::phaseName(flowPhaseToEbosPhaseIdx(phaseIdx));
 
                 if (std::isnan(mass_balance_residual[phaseIdx])
-                    || std::isnan(CNV[phaseIdx])
-                    || (phaseIdx < numPhases() && std::isnan(well_flux_residual[phaseIdx]))) {
+                    || std::isnan(CNV[phaseIdx])) {
                     OPM_THROW(Opm::NumericalProblem, "NaN residual for phase " << phaseName);
                 }
                 if (mass_balance_residual[phaseIdx] > maxResidualAllowed()
-                    || CNV[phaseIdx] > maxResidualAllowed()
-                    || (phaseIdx < numPhases() && well_flux_residual[phaseIdx] > maxResidualAllowed())) {
+                    || CNV[phaseIdx] > maxResidualAllowed()) {
                     OPM_THROW(Opm::NumericalProblem, "Too large residual for phase " << phaseName);
```

Comment:
I answered in the wrong place. I suggested a strategy to achieve this for the well model. Maybe we can do something similar to the BlackModelEbos too.